sdk/core/Azure.Core/src/ResponseClassifier.cs

```diff
@@ -54,8 +54,21 @@ public virtual bool IsRetriable(HttpMessage message, Exception exception)
         /// </summary>
         public virtual bool IsErrorResponse(HttpMessage message)
         {
-            var statusKind = message.Response.Status / 100;
-            return statusKind == 4 || statusKind == 5;
+            var status = message.Response.Status;
+            switch (status)
+            {
+                case 409:
```

Comment:
Should we consider applying this to 412 as well?   We see that with ETag mismatches during Storage operations.

Reply:
I think we should only apply this to 412.

Reply:
damn.   I was hoping Storage was doing something clever with the match headers in their 409 scenarios to help differentiate the expected cases.   The change to 412 is helpful, regardless.

Reply:
Ah, so I'm not crazy, it's just the storage not following the spec. Maybe it's all right to add both. The logic is limited to the conditional header use anyway.